catalog/spatial/geocoding/spatial-geocoding-lucene/src/main/java/org/codice/ddf/spatial/geocoding/index/GeoNamesCatalogIndexer.java

```diff
@@ -0,0 +1,189 @@
+      throws GeoEntryIndexingException, GeoEntryExtractionException,
+          GeoNamesRemoteDownloadException {
+    if (StringUtils.isBlank(resource)) {
+      LOGGER.debug("The resource was null or empty.");
+      return;
+    }
+
+    List<Metacard> metacards = new ArrayList<>();
+
+    final GeoEntryExtractor.ExtractionCallback extractionCallback =
+        new GeoEntryExtractor.ExtractionCallback() {
+          @Override
+          public void extracted(final GeoEntry newEntry) throws GeoEntryIndexingException {
+            Metacard metacard = transformGeoEntryToMetacard(newEntry);
+            if (metacard != null) {
+              metacards.add(metacard);
+            }
+          }
+
+          @Override
+          public void updateProgress(final int progress) {
+            if (progressCallback != null) {
+              progressCallback.updateProgress(progress);
+            }
+          }
+        };
+
+    geoEntryExtractor.pushGeoEntriesToExtractionCallback(resource, extractionCallback);
+
+    if (CollectionUtils.isEmpty(metacards)) {
+      LOGGER.debug("No Metacards were created from the resource.");
+      return;
+    }
+
+    double totalMetacards = metacards.size();
+    progressCallback.updateProgress(0);
+
+    for (int i = 0; i < totalMetacards; i += BATCH_SIZE) {
+
+      int lastIndex = i + BATCH_SIZE;
+      if (lastIndex > metacards.size()) {
+        lastIndex = metacards.size();
+      }
+
+      List<Metacard> sublist = metacards.subList(i, lastIndex);
+      Map<String, Serializable> properties = new HashMap<>();
+      CreateRequest createRequest = new CreateRequestImpl(sublist, properties);
+      try {
+        CreateResponse createResponse = executeCreate(createRequest);
+        List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
+        LOGGER.trace(
+            "Created {} metacards.", createdMetacards == null ? 0 : createdMetacards.size());
+      } catch (IngestException | SourceUnavailableException e) {
+        LOGGER.debug("Unable to create Metacards", e);
+      }
+      progressCallback.updateProgress((int) (i / totalMetacards));
+    }
+
+    progressCallback.updateProgress(100);
+  }
+
+  private CreateResponse executeCreate(CreateRequest createRequest)
+      throws IngestException, SourceUnavailableException {
+    return catalogFramework.create(createRequest);
+  }
+
+  @Override
+  public void updateIndex(
+      List<GeoEntry> newEntries, boolean create, ProgressCallback progressCallback)
+      throws GeoEntryIndexingException {}
+
+  @Override
+  public void updateIndex(
+      String resource,
+      GeoEntryExtractor geoEntryExtractor,
+      boolean create,
+      ProgressCallback progressCallback)
+      throws GeoEntryIndexingException, GeoEntryExtractionException,
+          GeoNamesRemoteDownloadException {
+    updateIndex(resource, geoEntryExtractor, progressCallback);
```

Comment:
Now that we're storing entries in metacards, do we not need to worry about deleting existing ones if `create` is true? In my PR I start by deleting all the country shapes in that scenario.

Reply:
updated to remove the previous metacards when create is true